### PR TITLE
tests: Fix flaky create_azure_snapshot_and_restore_with_secondary_endpoint

### PR DESCRIPTION
**Pull request by matriv**
Use host name instead of address, since the secondary endpoint must be
a prefix + the valid endpoint, and when this is an ip address
(`127.0.0.1`) instead of `localhost`, it seems to fail. Switching to
`localhost`, requires that the secondary endpoint is a subdomain of
`localhost`, therefore: `invalid.localhost` for the relevant test.

### CODE REVIEW

**Comment by matriv** [comment]
Will remove the `@Repeat` before merging.
Run locally with 100 iterations and seems to also work for CI.

**Comment by matriv** [comment]
retest this please

File: plugins/es-repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureSnapshotIntegrationTest.java [hunk 1]
```diff
@@ -166,11 +167,11 @@ public void test_invalid_settings_to_create_azure_repository() {
...
     private String httpServerUrl() {
         InetSocketAddress address = httpServer.getAddress();
-        return "http://" + InetAddresses.toUriString(address.getAddress()) + ":" + address.getPort();
+        return "http://" + address.getAddress().getCanonicalHostName() + ":" + address.getPort();
     }
 
     private String invalidHttpServerUrl() {
         InetSocketAddress address = httpServer.getAddress();
-        return "http://dummy.invalid:" + address.getPort();
+        return "http://invalid." + address.getAddress().getCanonicalHostName() + ":" + address.getPort();
```
Comment (seut) [inline plugins/es-repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureSnapshotIntegrationTest.java:175]: Is this change really related? As this will be invalid anyhow, I don't quite understand what this change improves...

**Review by seut — APPROVED** [review]
lgtm otherwise

File: plugins/es-repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureSnapshotIntegrationTest.java (hunk 1 above)
Comment (matriv) [reply plugins/es-repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureSnapshotIntegrationTest.java:175]: The secondary endpoint must be a subdomain of the primary.
So when the primary is an IP then this ^^ validation passes somehow, but then it seems for ci it frequently fails afterwards to connect to the http server.
With the change to `localhost` the secondary endpoint must be` <something>.localhost` to not trigger the validation, I will clarify it in the commit msg/PR description.

thank you!
